### PR TITLE
overlord/snapstate: check changes to SnapState for conflicts also.

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -153,7 +154,7 @@ func doInstall(s *state.State, snapst *SnapState, ss *SnapSetup) (*state.TaskSet
 	return state.NewTaskSet(tasks...), nil
 }
 
-func checkChangeConflict(s *state.State, snapName string) error {
+func checkChangeConflict(s *state.State, snapst *SnapState, snapName string) error {
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go]: I would switch the order of params, is a bit strange the nil in the middle in many cases

Comment (pedronis) [inline overlord/snapstate/snapstate.go]: I mean  s, snapName, snapst

**Comment by pedronis** [comment]
LGTM, small note about param order, thank you

File: overlord/snapstate/snapstate.go (hunk 1 above)
Comment (chipaca) [inline overlord/snapstate/snapstate.go]: yeah... ok. I dithered about that :-) will do.

**Comment by mvo5** [comment]
Code looks fine, but a small comment why we are doing this would be nice (its also slightly misleading as this is not a change conflict).

**Pull request merged**